quantlib/util/compat.pyx

```diff
@@ -0,0 +1,32 @@
+""" Compatibility layer for Py2/Py3 support. """
+
+from cpython cimport PyBytes_AsString, PY_MAJOR_VERSION
+from libcpp.string cimport string
+
+cdef string py_compat_str_as_utf8_string(text):
```

Comment:
Suggestion: can these functions be named according to the `X_from_Y` convention, so that calling them looks like `x = X_from_Y(y)`, rather than `x = Y_as_X(y)`? While I was reading the PR I often had to stop and pause to figure out whether the argument was a Python string or a C++ string.